js/client-battle.js

```diff
@@ -421,8 +424,21 @@
 			}
 			return '<button name="openTimer" class="button timerbutton' + timerTicking + '"><i class="fa fa-hourglass-start"></i> ' + time + '</button>';
 		},
-		updateTimer: function () {
-			this.$('.timerbutton').replaceWith(this.getTimerHTML());
+		timerInterval: 0,
+		updateTimer: function (ticking) {
+			if (ticking && this.battle.kickingInactive > 1) this.battle.kickingInactive--;
+			var $timerButton = this.$('.timerbutton');
+			if ($timerButton.length) {
+				$timerButton.replaceWith(this.getTimerHTML());
+			} else {
+				this.clearInterval(this.timerInterval);
+				this.timerInterval = 0;
+			}
+		},
+		createTimer: function () {
+			if (this.timerInterval) clearInterval(this.timerInterval);
+			this.timerInterval = setInterval(this.updateTimer.bind(this, true), 1000);
```

Comment:
Also, put all this in `getTimerHTML`, there's no reason to add yet another function to the stack for something this simple.